### ISSUE TITLE
Creating camel-k with helm cannot publish integrations

### ISSUE THREAD

**Issue by longhai-he** [apache/camel-k#5888]
### What happened?
```
 Message:               integration kit default/kit-cs4buhs8rlr14be14ob0 is in state "Error". Failure: failure while building project: : signal: killed
    Reason:                IntegrationKitAvailable
    Status:                False
    Type:                  IntegrationKitAvailable
    Last Transition Time:  2024-10-11T06:31:48Z
    Last Update Time:      2024-10-11T06:31:48Z
    Message:               error during trait customization: security-context trait execution failed: could not find any integration deployment for my-it
```
### Steps to reproduce
```
1.$ helm repo add camel-k https://apache.github.io/camel-k/charts/
```
```
2.$ helm install camel-k [--set platform.build.registry.address=<my-registry>] camel-k/camel-k --force
```
```
3.create my-integration.yaml 
apiVersion: camel.apache.org/v1
kind: Integration
metadata:
  name: my-it
spec:
  flows:
  - from:
      parameters:
        period: "1000"
      steps:
      - setBody:
          simple: Hello Camel from ${routeId}
      - log: ${body}
      uri: timer:yaml
```
```
4.$ kubectl  apply -f my-integration.yaml 
```
### Relevant log output
```
Events:
  Type    Reason                       Age    From                                Message
  ----    ------                       ----   ----                                -------
  Normal  IntegrationConditionChanged  7m46s  camel-k-integration-controller      Condition "IntegrationPlatformAvailable" is "True
" for Integration my-it: default/camel-k
  Normal  IntegrationConditionChanged  7m46s  camel-k-integration-controller      Condition "TraitInfo" is "True" for Integration m
y-it: Applied traits: platform,camel,deployer
  Normal  IntegrationPhaseUpdated      7m46s  camel-k-integration-controller      Integration "my-it" in phase "Initialization"
  Normal  IntegrationConditionChanged  7m46s  camel-k-integration-controller      Condition "TraitInfo" is "True" for Integration m
y-it: Applied traits: init,camel,dependencies,deployer,gc,container,owner
  Normal  IntegrationPhaseUpdated      7m46s  camel-k-integration-controller      Integration "my-it" in phase "Building Kit"
  Normal  IntegrationConditionChanged  7m46s  camel-k-integration-controller      Condition "TraitInfo" is "True" for Integration m
y-it: Applied traits: camel,deployer,quarkus
  Normal  IntegrationConditionChanged  7m46s  camel-k-integration-controller      Condition "IntegrationKitAvailable" is "False" fo
r Integration my-it: creating a new integration kit
  Normal  ReasonRelatedObjectChanged   7m46s  camel-k-integration-kit-controller  Integration Kit "kit-cs4k78nj43qa5i1jti1g", creat
ed by Integration "my-it", changed phase to "Build Submitted"
  Normal  ReasonRelatedObjectChanged   7m46s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Scheduling"
  Normal  ReasonRelatedObjectChanged   7m46s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Pending"
  Normal  ReasonRelatedObjectChanged   7m46s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Running"
  Normal  ReasonRelatedObjectChanged   7m46s  camel-k-integration-kit-controller  Integration Kit "kit-cs4k78nj43qa5i1jti1g", creat
ed by Integration "my-it", changed phase to "Build Running"
  Normal  ReasonRelatedObjectChanged   2m46s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Failed"
  Normal  ReasonRelatedObjectChanged   2m41s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Initialization" (recovery 1 of 5)
  Normal  ReasonRelatedObjectChanged   2m41s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Scheduling" (recovery 1 of 5)
  Normal  ReasonRelatedObjectChanged   2m41s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Pending" (recovery 1 of 5)
  Normal  ReasonRelatedObjectChanged   2m41s  camel-k-build-controller            Build "kit-cs4k78nj43qa5i1jti1g", created by Inte
gration "my-it", changed phase to "Running" (recovery 1 of 5)
```
#############camel-k operator logs ########
```
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.controller.build","msg":"State transition","request-namespace":"defau
lt","request-name":"kit-cs4k78nj43qa5i1jti1g","api-version":"camel.apache.org/v1","kind":"Build","ns":"default","name":"kit-cs4k78n
j43qa5i1jti1g","phase-from":"Pending","phase-to":"Running"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.builder","msg":"running builder task builder in context directory: /t
mp/kit-cs4k78nj43qa5i1jti1g-602271261"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"Executed command: /usr/bin/cp --recursive /usr/sh
are/maven/mvnw/. ."}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.controller.integrationkit","msg":"Invoking action build","request-nam
espace":"default","request-name":"kit-cs4k78nj43qa5i1jti1g","api-version":"camel.apache.org/v1","kind":"IntegrationKit","ns":"defau
lt","name":"kit-cs4k78nj43qa5i1jti1g"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.controller.integrationkit","msg":"Build running","request-namespace":
"default","request-name":"kit-cs4k78nj43qa5i1jti1g","api-version":"camel.apache.org/v1","kind":"IntegrationKit","ns":"default","nam
e":"kit-cs4k78nj43qa5i1jti1g"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven","msg":"executing: ./mvnw -V --no-transfer-progress -Dstyle.col
or=never package -Dmaven.test.skip=true -Dmaven.repo.local=/etc/maven/m2 --global-settings /tmp/kit-cs4k78nj43qa5i1jti1g-602271261/
maven/settings.xml","MAVEN_OPTS":""}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"Executed command: ./mvnw -V --no-transfer-progres
s -Dstyle.color=never package -Dmaven.test.skip=true -Dmaven.repo.local=/etc/maven/m2 --global-settings /tmp/kit-cs4k78nj43qa5i1jti
1g-602271261/maven/settings.xml"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"Apache Maven 3.8.6 (84538c9988a25aec085021c365c56
0670ad80f63)"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"Maven home: /usr/share/maven/wrapper/dists/apache
-maven-3.8.6-bin/5f1464e3/apache-maven-3.8.6"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"Java version: 17.0.12, vendor: Eclipse Adoptium, 
runtime: /opt/java/openjdk"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"Default locale: en_US, platform encoding: UTF-8"}
{"level":"info","ts":"2024-10-11T15:36:31Z","logger":"camel-k.maven.build","msg":"OS name: \"linux\", version: \"3.10.0-1160.92.1.e
l7.x86_64\", arch: \"amd64\", family: \"unix\""}
{"level":"info","ts":"2024-10-11T15:36:32Z","logger":"camel-k.maven.build","msg":"Scanning for projects..."}
{"level":"info","ts":"2024-10-11T15:36:32Z","logger":"camel-k.maven.build","msg":""}
{"level":"info","ts":"2024-10-11T15:36:32Z","logger":"camel-k.maven.build","msg":"---------< org.apache.camel.k.integration:camel-k
-integration >---------"}
{"level":"info","ts":"2024-10-11T15:36:32Z","logger":"camel-k.maven.build","msg":"Building camel-k-integration 2.4.0"}
{"level":"info","ts":"2024-10-11T15:36:32Z","logger":"camel-k.maven.build","msg":"--------------------------------[ jar ]----------
-----------------------"}
```
################camel-k it##################
```
[root@gykg-gksk-sz01p-ecs-uat-jkpt-ccenode-0930030 camel-k]# kubectl get it
NAME    PHASE          READY   RUNTIME PROVIDER   RUNTIME VERSION   KIT                        REPLICAS
my-it   Building Kit           quarkus            3.8.1             kit-cs4k78nj43qa5i1jti1g
```

### Camel K version

v.2.4.0

**Comment by squakez** [comment]
Hello, thanks for reporting. It seems there is some problem when trying to build the maven project. According to the log provided, the build starts successfully, so, it's the publishing part to the registry that may be failing. You can have a look at the [Camel K troubleshooting guide to find more useful information](https://camel.apache.org/camel-k/next/troubleshooting/troubleshooting.html).
In order to help, it would be nice to get the entire output log of the operator and additionally the output of the Build custom resource (ie, `kubectl get build xyz -o yaml`). If there is some problem to the registry (like it typically happens), you can also check the IntegrationPlatform status (ie, `kubectl get itp -o yaml`).

**Comment by squakez** [comment]
Hello. Is the problem fixed now? should we close this issue?

**Comment by longhai-he** [comment]
I am very sorry, there is no error at present, because I did not see the specific error problem. I guess it was the private domain network problem that caused the failure of pulling jar packages when camel-k operator s used maven to build, but I have not found the specific reason

**Comment by kanaiyathacker1980** [comment]
Hi,

I encountered a similar error:


{"level":"error","ts":"2025-07-31T08:03:40Z","msg":"Reconciler error","controller":"integration-controller","controllerGroup":"camel.apache.org","controllerKind":"Integration","Integration":{"name":"sqs-to-knative","namespace":"default"},"namespace":"default","name":"sqs-to-knative","reconcileID":"e96e3b7b-4248-4f90-818a-f019fb1ef149","error":"error during trait customization: security-context trait execution failed: could not find any integration deployment for sqs-to-knative","stacktrace":""}

I tried all possible ways to fix it, but in the end, I had to use the kamel reset command. However, I can't use this in production as it wipes out all integrations. I'm currently using Camel K version 2.7.0.

What’s the best way to resolve this issue?
